src/codelabs/iterables/index.md

```diff
@@ -0,0 +1,571 @@
+to convert all `int` to `String` as you can see in the example below.
+
+{% prettify dart %}
+  Iterable<String> output = numbers.map((number) => number.toString());
+{% endprettify %}
+
+{{site.alert.note}}
+  `map` returns a "lazy" `Iterable`, meaning that the supplied function will
+  only be called when the elements are iterated.
+{{site.alert.end}}
+
+### Example: Using map to change elements
+
+Run this example to see how `map` can be used to multiply all the elements
+of an `Iterable` by two. What do you think the output will be?
+
+[//]: https://gist.github.com/ba4352e121b19ecb1b171cce553de72f
+<iframe
+  src="{{site.dartpad-embed}}?id=ba4352e121b19ecb1b171cce553de72f"
+  style="border: 1px solid lightgrey; margin-top: 10px; margin-bottom: 25px"
+  frameborder="no"
+  height="200"
+  width="100%" >
+</iframe>
+
+### Exercise: Mapping to a different type
+
+In the previous example you multiplied the elements of an `Iterable` by two.
+Both the input and the output of that operation were an `Iterable` of `int`.
+
+In this exercise, you code takes an `Iterable` of `User`, and you need to
+return an `Iterable` that contains the `age` multiplied by two.
+
+As a reminder, the class `User` contains a member `age` of the type `int`.
+
+{% prettify dart %}
+  class User {
+    String name;
+    int age;
+  }
+{% endprettify %}
+
+[//]: https://gist.github.com/c2ca9085b2376471b30569d570590894
+<iframe
+src="{{site.dartpad-embed}}?id=c2ca9085b2376471b30569d570590894&theme=dark"
+frameborder="no"
+height="325"
+width="100%" >
+</iframe>
+
+{{site.alert.secondary}}
+  **Quick review:**
+  * `map` applies a function to all the elements of an `Iterable`.
+  * The output of `map` is another `Iterable`.
+  * The function is not evaluated until the `Iterable` is iterated.
+{{site.alert.end}} 
+
+## Exercise: Putting it all together
+
+It’s time to practice what you’ve learned in one final exercise.
+
+This exercise provides the class `Email` with a constructor, that takes a `String`.
+
+{% prettify dart %}
+class Email {
+  String address;
+
+  Email(
+    this.address,
+  );
+}
+{% endprettify %}
+
+This exercise also provides the function `checkIfEmailInvalid` that returns `true` if
+an `Email` is invalid.
+
+|--------------------------+-----------------------------------+-------------|
+| Function                 | Type signature                    | Description |
+|--------------------------|-----------------------------------|-------------|
+| checkIfEmailInvalid()    | `bool checkIfEmailInvalid(Email)` | Returns `true` if the provided `Email` is invalid. | 
+|--------------------------+-----------------------------------+-------------|
+{:.table .table-striped}
+
+Your goal is:
+
+**Part 1:** `parseEmails`
+- Write the function `parseEmails` that takes an `Iterable` of `String` containing emails, 
+and returns an `Iterable` of `Email`.
+- Use the method `map` to map between the `String` to `Email`.
```

Comment:
[nit] I wouldn't say "between" since it is one direction. How about "to map from a `String` to an `Email`"?